draft-ietf-httpbis-no-vary-search.md

```diff
@@ -390,6 +392,15 @@ So, for example, given any non-default value for `No-Vary-Search`, such as `No-V
   <dd><tt>+</tt> and <tt>%20</tt> are both parsed as U+0020 SPACE</dd>
 </dl>
 
+# Caching
+
+{:aside}
+> A more thorough discussion of how the above algorithms are used in HTTP caches {{HTTP-CACHING}} generally is likely merited.
+
+Servers should send no more than one distinct non-empty value for the `No-Vary-Search` field in response to requests for a given pathname. Cache implementations may fail to reuse a stored response whose target URI is not equal to the presented target URI ({{Section 7.1 of HTTP}}) even though the two are equivalent modulo URL search variance for the variance obtained ({{obtain-a-url-search-variance}}) from the stored response, if the cache has more recently stored a response with a target URI which is equal, excluding the query, and a different non-empty value for the `No-Vary-Search` field.
```

Comment:
The final sentence here is so long. It is very hard to follow. And as noted, it's kind of modifying a caching integration that doesn't exist.

My attempt at clearing this up would be something like the following:

This specification modifies the "Constructing Responses from Caches" section of RFC 9911 by replacing the statement that a cache must not reuse a stored response unless "the presented target URI and that of the stored response match". Instead, cache implementations use the condition that **the cache URIs match** given the presented target URI and the stored response.

The **cache URIs match**, given a URL |url| and a stored response |response|, if the following algorithm returns true:

1. An implementation may perform the following step:
   1. If all of the following are true:
      - The cache has more recently stored a response |moreRecentResponse| with a target URI which is equal, excluding the query, to |url|;
      - |moreRecentResponse|'s `No-Vary-Search` field is non-empty; and
      - |moreRecentResponse|'s `No-Vary-Search` field does not equal |response|'s `No-Vary-Search` field,
     
      then return true if |response|'s target URI equals |url|. Otherwise return false.
1. Let |urlSearchVariance| be the result of obtaining a URL search variance given |response|.
1. Return whether |url| is equivalent modulo URL search variance to |response|'s target URI, given |urlSearchVariance|.

This only resolves my confusion with the long sentence, and doesn't resolve my confusion with how this actually helps the implementation. (See below.)

Reply:
Alright, explicitly integrated with RFC 9111. Not as bad as I'd feared, mostly because it just lists cases where a cache MUST NOT reuse a stored response (and therefore implies that everything else MAY be reused) -- at least, from my quick read.

I've made this a bit more descriptive than algorithmic but hopefully still respecting the spirit of your comment. PTAL.